test/e2e/main/main_test.go

```diff
@@ -817,3 +818,73 @@ func throwHTTPError(resp *http.Response) {
 	errorBytes, _ := ioutil.ReadAll(resp.Body)
 	panic(fmt.Errorf("unexpected HTTP status code (%d): %s", resp.StatusCode, string(errorBytes)))
 }
+
+func TestCacheCR(t *testing.T) {
+	spec := tutils.DefaultSpec(testKube)
+	name := strcase.ToKebab(t.Name())
+	spec.Name = name
+	testKube.CreateInfinispan(spec, tutils.Namespace)
+	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, spec.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
+	cacheNameTemplateName := "testtemplatename"
+	cacheNameTemplateXML := "testtemplatexml"
+
+	ispn := ispnv1.Infinispan{}
+	testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Namespace: spec.Namespace, Name: spec.Name}, &ispn)
```

Comment:
Seems to we don't need to re-query Infinispan CR again.

Reply:
@dmvolod can you please review the changes. I hope I understood the comment and made correct changes.

Reply:
@durgeshanaokar please see suggestion changes below

Reply:
I will also create a separate issue, to optimize `testKube.WaitForInfinispanCondition` this should be better to return latest Infinispan CR from it and remove redundant Get before auth for all cases.
@durgeshanaokar you are welcome to fix this issue.

Reply:
@dmvolod sure it will be great opportunity and learning for me :)